### PR TITLE
test: upload xml reports to test account's obj storage

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

Storing test executions from dev/main branch to test account's object storage

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

File: .github/workflows/e2e-suite.yml [hunk 1]
```diff
@@ -36,6 +36,18 @@ jobs:
...
+            linode-cli obj --cluster us-southeast-1 put "${report_filename}" test-report-bucket
+          else
+            echo "Tests failed, but attempting to upload results anyway"
+            linode-cli obj --cluster us-southeast-1 put "${report_filename}" test-report-bucket
+            exit 1
+          fi
         env:
           LINODE_CLI_TOKEN: ${{ secrets.LINODE_TOKEN }}
+          LINODE_CLI_OBJ_ACCESS_KEY: ${{ secrets.LINODE_CLI_OBJ_ACCESS_KEY }}
+          LINODE_CLI_OBJ_SECRET_KEY: ${{ secrets.LINODE_CLI_OBJ_SECRET_KEY }}
```
Comment (jriddle-linode) [inline .github/workflows/e2e-suite.yml:53]: Is this a duplicate or just github?

Also might be worth a rebase.

**Review by jriddle-linode — APPROVED** [review]

File: .github/workflows/e2e-suite.yml (hunk 1 above)
Comment (lgarber-akamai) [inline .github/workflows/e2e-suite.yml:53]: I wonder if we could dynamically generate these using `linode-cli obj regenerate-keys` to skip adding them as GitHub secrets.

File: .github/workflows/e2e-suite.yml [hunk 2]
```diff
@@ -36,6 +36,18 @@ jobs:
...
+          timestamp=$(date +'%Y%m%d%H%M')
+          report_filename="${timestamp}_cli_test_report.xml"
+          
+          if pytest -m tests/integration --disable-warnings --junitxml="${report_filename}"; then
+            linode-cli obj --cluster us-southeast-1 put "${report_filename}" test-report-bucket
+          else
+            echo "Tests failed, but attempting to upload results anyway"
+            linode-cli obj --cluster us-southeast-1 put "${report_filename}" test-report-bucket
+            exit 1
+          fi
```
Comment (lgarber-akamai) [inline .github/workflows/e2e-suite.yml:49]: Small stylistic suggestion, feel free to ignore:
```suggestion
          timestamp=$(date +'%Y%m%d%H%M')
          report_filename="${timestamp}_cli_test_report.xml"
          status=0

          if ! pytest -m tests/integration --disable-warnings --junitxml="${report_filename}"; then
            echo "Tests failed, but attempting to upload results anyway"
            status=1
          fi

          linode-cli obj --cluster us-southeast-1 put "${report_filename}" test-report-bucket
          exit ${status}
```

**Review by lgarber-akamai — APPROVED** [review]
Seems to work well on my end, nice work!

**Pull request merged**